upp/classes/components.py

```diff
@@ -64,15 +64,23 @@ def check_num_jets(self, num_jets, sampling_frac=None, cuts=None, silent=False):
         # check with tolerance to avoid failure midway through preprocessing
         if available < num_jets * 1.01:
             raise ValueError(
-                f"{num_jets:,} jets requested, but only {total:,} are estimated to be in"
-                f" {self}. With a sampling fraction of {sampling_frac}, at most {available:,} of"
-                " these are available. You can either reduce the number of requested jets or"
-                " increase the sampling fraction."
+                f"{num_jets:,} jets requested, but only {total:,} are estimated to be"
+                f" in {self}. With a sampling fraction of {sampling_frac}, at most"
+                f" {available:,} of these are available. You can either reduce the"
+                " number of requested jets or increase the sampling fraction."
             )
 
         if not silent:
             log.info(f"Estimated {available:,} {self} jets available - {num_jets:,} requested")
 
+    def get_auto_sampling_frac(self, num_jets, cuts=None, silent=False):
+        total = self.reader.estimate_available_jets(cuts, self.num_jets_estimate)
+        auto_sampling_frac = num_jets / (total * 1.01)  # 1.01 is a tolerance
```

Comment:
Is the tolerance here applied in the wrong direction? I used it before to slightly underestimate the total number of jets to be conservative, here it looks like the opposite